core/vm/ewasm_eei.go

```diff
@@ -0,0 +1,842 @@
+		in.contract.Gas += GasCostCallStipend
+		return ErrEEICallFailure
+	}
+
+	// TODO tracing
+	// TODO check that EIP-150 is respected
+
+	// Add amount to recipient
+	in.evm.Transfer(in.StateDB, contract.Address(), addr, value)
+
+	// Load the contract code in a new VM structure
+	targetContract := NewContract(contract, AccountRef(addr), value, uint64(gas))
+	code := in.StateDB.GetCode(addr)
+	targetContract.SetCallCode(&addr, in.StateDB.GetCodeHash(addr), code)
+
+	return callCommon(in, contract, targetContract, input, value, snapshot, gas)
+}
+
+func storageStore(p *exec.Process, interpreter *InterpreterEWASM, pathOffset int32, valueOffset int32) {
+	loc := common.BytesToHash(readU256(p, pathOffset))
+	val := common.BytesToHash(readU256(p, valueOffset))
+
+	fmt.Println(val, loc)
+	nonZeroBytes := 0
+	for _, b := range val.Bytes() {
+		if b != 0 {
+			nonZeroBytes++
+		}
+	}
+
+	oldValue := interpreter.StateDB.GetState(interpreter.contract.Address(), loc)
+	oldNonZeroBytes := 0
+	for _, b := range oldValue.Bytes() {
+		if b != 0 {
+			oldNonZeroBytes++
+		}
+	}
+
+	if (nonZeroBytes > 0 && oldNonZeroBytes != nonZeroBytes) || (oldNonZeroBytes != 0 && nonZeroBytes == 0) {
+		interpreter.gasAccounting(GasCostSSet)
+	} else {
+		// Refund for setting one value to 0 or if the "zeroness" remains
+		// unchanged.
+		interpreter.gasAccounting(GasCostSReset)
+	}
+
+	interpreter.StateDB.SetState(interpreter.contract.Address(), loc, val)
+}
+
+func storageLoad(p *exec.Process, interpreter *InterpreterEWASM, pathOffset int32, resultOffset int32) {
+	interpreter.gasAccounting(interpreter.gasTable.SLoad)
+	loc := common.BytesToHash(readU256(p, pathOffset))
+	valBytes := interpreter.StateDB.GetState(interpreter.contract.Address(), loc).Bytes()
+	p.WriteAt(valBytes, int64(resultOffset))
+}
+
+func getCaller(p *exec.Process, in *InterpreterEWASM, resultOffset int32) {
+	callerAddress := in.contract.CallerAddress
+	in.gasAccounting(GasCostBase)
+	p.WriteAt(callerAddress.Bytes(), int64(resultOffset))
+}
+
+func getCallValue(p *exec.Process, in *InterpreterEWASM, resultOffset int32) {
+	in.gasAccounting(GasCostBase)
+	p.WriteAt(swapEndian(in.contract.Value().Bytes()), int64(resultOffset))
+}
+
+func codeCopy(p *exec.Process, in *InterpreterEWASM, resultOffset int32, codeOffset int32, length int32) {
+	in.gasAccounting(GasCostVeryLow + GasCostCopy*(uint64(length+31)>>5))
+	code := in.contract.Code
+	p.WriteAt(code[codeOffset:codeOffset+length], int64(resultOffset))
+}
+
+func getCodeSize(p *exec.Process, in *InterpreterEWASM) int32 {
+	in.gasAccounting(GasCostBase)
+	code := in.StateDB.GetCode(*in.contract.CodeAddr)
+	return int32(len(code))
+}
+
+func getBlockCoinbase(p *exec.Process, in *InterpreterEWASM, resultOffset int32) {
+	in.gasAccounting(GasCostBase)
+	p.WriteAt(in.evm.Coinbase.Bytes(), int64(resultOffset))
+}
+
+// Unline EVM1, Gas costs do not depend on the amount of gas / 64.
```

Comment:
What does this mean exactly?

Reply:
It means that the bytes of the coinbase are copied to the VM memory address pointed by `resultOffset`.

Reply:
Still not any more clearer :)